test/ruby/test_thread_queue.rb

```diff
@@ -585,9 +585,14 @@ def test_queue_close_multi_multi
     count_items = rand(3000..5000)
     count_producers = rand(10..20)
 
+    # ensure threads do not start running too soon and complete before we check status
+    mutex = Mutex.new
+    mutex.lock
+
     producers = count_producers.times.map do
       Thread.new do
-        sleep(rand / 100)
```

Comment:
Any idea why there was a random sleep and no mutex?

Reply:
It was probably sufficient to make sure the threads did not start running too early on CRuby.